src/port1.0/portlint.tcl

```diff
@@ -249,6 +249,30 @@ proc portlint::lint_checksum {checksum_string} {
     return [list $errors $warnings]
 }
 
+# lint_platforms
+#
+# Checks a given Portfile platforms string.  Returns a list of error strings.
+#
+# Returns an empty list if no issues are found.
+proc portlint::lint_platforms {platforms} {
+    set issues [list]
+
+    global lint_platforms ports_lint_nitpick
+
+    foreach platform $platforms {
+        set platname [lindex $platform 0]
+        if {$platname ni $lint_platforms} {
+            lappend issues "Unknown platform: $platname"
+        }
+    }
+
+    if {[tbool ports_lint_nitpick] && [llength $platforms] == 1 && "darwin" in $platforms} {
```

Comment:
`[llength $platforms] == 1 && "darwin" in $platforms` could be simplified to `$platforms eq "darwin"`.